### DISCUSSION TITLE
Any alternative to Sevilla's Soft MIDI USB-USB device?

### DISCUSSION THREAD

**Discussion by tutseryday** [onlaj/Piano-LED-Visualizer#405] [General]
Costs quite a bit with shipping to my country. Any alternatives to this? I have tired RTP and just couldn't make it work because my Pi kept getting disconnected over and over again. RTPMidi kept giving me errors saying "my.pi.ip has not answered the connection request", same with bluetooth.

**Comment by onlaj** [comment]
This is alternative: https://gist.github.com/gbaman/50b6cca61dd1c3f88f41
Basically, you transform your RPi into a MIDI device and use another device, such as a PC, to transmit data between the RPi and piano.

Unfortunately, my RPi has a damaged USB port, and I couldn't find a new unit for a reasonable price, so I haven't tested it yet.

edit. and it works only with RPi Zero

**Comment by psscin01** [comment]
There might be a cheap DIY alternative using 2 QTPY samd21 boards. [https://github.com/okyeron/2Host](url)

I am trying to use two qtpy rp2040 with no success so far, but he is going to look at the code this weekend and get back to me. I will let you know if this works. If so, the cost is something around USD: $9.00.

**Comment by andre2007** [comment]
I wonder whether it will work, to just connect these 2 products together:

![image](https://github.com/user-attachments/assets/113886b1-ff3d-4b29-8255-b0667e10de35)

(midi female to midi female)
![image](https://github.com/user-attachments/assets/cc4727a6-9b9a-4bdd-9a64-551cb742acb6)

Or will this not work?

**Reply by onlaj** [reply]
I tried this, it didn't work. I used some cheap aliexpress controllers, with more expensive one it would probably work just fine. But then it would cost as much or more as Sevilla's device.

**Reply by andre2007** [reply]
> I tried this, it didn't work. I used some cheap aliexpress controllers, with more expensive one it would probably work just fine. But then it would cost as much or more as Sevilla's device.

Thanks for the information. I contacted the Sevilla shop support, because it looks like the shop is no longer active and there is no possibility to buy the device anywhere. I would like to buy any solution which is in the same price range as the Sevilla device, but was not able to find any.

In the meantime I will try the software solution described in this blog, maybe it is working:
https://neuma.studio/raspberry-pi-as-usb-bluetooth-midi-host/

Kind regards
André

**Comment by psscin01** [comment]
Hey! Thanks for this. I will look into it, but I haven't had much luck with
those types of connectors.

On Fri, Sep 20, 2024 at 2:50 PM andre2007 ***@***.***> wrote:

> I wonder whether it will work, to just connect these 2 products together:
>
> image.png (view on web)
> <https://github.com/user-attachments/assets/113886b1-ff3d-4b29-8255-b0667e10de35>
>
> (midi female to midi female)
> image.png (view on web)
> <https://github.com/user-attachments/assets/cc4727a6-9b9a-4bdd-9a64-551cb742acb6>
>
> Or will this not work?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/onlaj/Piano-LED-Visualizer/discussions/405#discussioncomment-10708508>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ASWOLIUSBT43EV4UDYZVFDLZXRVABAVCNFSM6AAAAABOSVPO6SVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANZQHA2TAOA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>


-- 
Dr. Parker Scinta, DMA
Kentucky Country Day
Director of Student Activities
Community Service Coordinator

***@***.***
4100 Springdale Rd.
Louisville, KY 40241
Pronouns: he/him/his

**Comment by psscin01** [comment]
I’m working on trying to reverse engineer the Sevilla device but I just
haven’t had much time. If anyone has successfully done this I’d love to
hear about it.

Dr. Parker Scinta, DMA
Kentucky Country Day
Director of Student Activities
Community Service Coordinator

***@***.***
4100 Springdale Rd.
Louisville, KY 40241
Pronouns: he/him/his

**Comment by andre2007** [comment]
@onlaj 
I finally got it working without any additional device 🎉. I am now able to connect the raspberry pi zero directly to my windows computer just with a default usb cable (even not OTG adapter needed).

This does the trick:

Just add this to /boot/config.txt
```
dtoverlay=dwc2
```

and load these modules `dwc2` and `g_midi`. The raspberry pi will be listed in in Synthesia as "midi function".
```
"dwc2" | sudo tee -a /etc/modules
"g_midi" | sudo tee -a /etc/modules
```
